File: app/src/main/java/com/shafa/ali/kavir_msg/adapters/CommentAdapter.java
package com.shafa.ali.kavir_msg.adapters;

import android.content.Context;
import android.os.Build;
import android.support.v7.widget.RecyclerView;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.shafa.ali.kavir_msg.R;
import com.shafa.ali.kavir_msg.models.CommentModel;

import java.util.List;

public class CommentAdapter extends RecyclerView.Adapter<CommentAdapter.Holder> {
    private List<CommentModel> commentModelList;
    private Context context;

    public CommentAdapter(List<CommentModel> commentModelList, Context context) {
        this.commentModelList = commentModelList;
        this.context = context;
    }

    @Override
    public Holder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_comment,parent,false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(Holder holder, int position) {
        CommentModel commentModel = commentModelList.get(position);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            holder.titleTv.setText(Html.fromHtml(commentModel.getName(),Html.FROM_HTML_MODE_COMPACT));
            holder.contentTv.setText(Html.fromHtml(commentModel.getContent(),Html.FROM_HTML_MODE_COMPACT));
        } else {
            holder.titleTv.setText(Html.fromHtml(commentModel.getName()));
            holder.contentTv.setText(Html.fromHtml(commentModel.getContent()));
        }
        holder.dateTv.setText(commentModel.getDate());
    }


    @Override
    public int getItemCount() {
        return commentModelList.size();
    }

    class Holder extends RecyclerView.ViewHolder{
        private TextView titleTv,contentTv,dateTv;

        public Holder(View itemView) {
            super(itemView);
            titleTv = (TextView)itemView.findViewById(R.id.title_comment_card);
            contentTv =(TextView)itemView.findViewById(R.id.content_comment_card);
            dateTv =(TextView)itemView.findViewById(R.id.date_comment_card);
        }
    }
}

File: app/src/main/java/com/kavirelectronic/ali/kavir_info/adapters/CommentAdapter.kt
package com.kavirelectronic.ali.kavir_info.adapters

import android.content.Context
import android.os.Build
import android.support.v7.widget.RecyclerView
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.kavirelectronic.ali.kavir_info.R
import com.kavirelectronic.ali.kavir_info.models.CommentModel
import com.kavirelectronic.ali.kavir_info.utility.FormatHelper

class CommentAdapter(private val commentModelList: List<CommentModel>, private val context: Context) : RecyclerView.Adapter<CommentAdapter.Holder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_comment, parent, false)
        return Holder(view)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val commentModel = commentModelList[position]
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            holder.titleTv.text = Html.fromHtml(commentModel.name, Html.FROM_HTML_MODE_COMPACT)
            holder.contentTv.text = Html.fromHtml(commentModel.content, Html.FROM_HTML_MODE_COMPACT)
        } else {
            holder.titleTv.text = Html.fromHtml(commentModel.name)
            holder.contentTv.text = Html.fromHtml(commentModel.content)
        }
        holder.dateTv.text = FormatHelper.toPersianNumber(commentModel.date)
    }

    override fun getItemCount(): Int {
        return commentModelList.size
    }

     inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val titleTv: TextView
        val contentTv: TextView
        val dateTv: TextView

        init {
            titleTv = itemView.findViewById<View>(R.id.title_comment_card) as TextView
            contentTv = itemView.findViewById<View>(R.id.content_comment_card) as TextView
            dateTv = itemView.findViewById<View>(R.id.date_comment_card) as TextView
        }
    }

}